src/lib/error.ts

```diff
@@ -9,11 +9,19 @@ export enum ErrorCode {
   inclusionPhaseNotInProgress = "inclusion_phase_not_in_progress",
   inclusionAlreadyInProgress = "inclusion_already_in_progress",
   invalidParamsPassedToCommand = "invalid_params_passed_to_command",
+  noLongerSupported = "no_longer_supported",
 }
 
 export class BaseError extends Error {
   // @ts-ignore
   errorCode: ErrorCode;
+
+  constructor(message?: string) {
+    super(message);
+    // We need to set the prototype explicitly
+    Object.setPrototypeOf(this, BaseError.prototype);
+    Object.getPrototypeOf(this).name = "ZWaveError";
```

Comment:
```suggestion
    Object.getPrototypeOf(this).name = "BaseError";
```
but this would require overriding it in the subclasses too. If you don't care about having the error class name in stack traces, e.g.
```
NoLongerSupportedError: This is an error message
  at foo.js:80
  at bar.js:81
```
then we can just omit that line.